Assets/_BForBoss/_Entities/Scripts/Navigation/AgentNavigationBehaviour.cs

```diff
@@ -41,11 +40,33 @@ public void MovementUpdate()
             }
         }
 
+        public void PauseNavigation()
+        {
+            if (_agent.enabled)
+            {
+                _agent.isStopped = true;
+                _agent.enabled = false;
+            }
+        }
+        
+        public void ResumeNavigation()
+        {
+            if (!_agent.enabled)
+            {
+                _agent.enabled = true;
+                if (_agent.isOnNavMesh)
+                {
+                    _agent.isStopped = false;
+                }
+            }
+        }
+
         private void Awake()
         {
             _agent = GetComponent<NavMeshAgent>();
             if (_agent == null)
             {
+                //Do we need this? the component has a RequireComponent flag on it already
```

Comment:
Fair point, maybe this was added just in case the `RequireComponent` attribute gets removed in the future? I'm down for removing this too though

Reply:
Just remove it. Overlooked code